pkg/controllers/management/k3supgrade/deployPlans.go

```diff
@@ -0,0 +1,168 @@
+package k3supgrade
+
+import (
+	"fmt"
+	"reflect"
+
+	planv1 "github.com/rancher/system-upgrade-controller/pkg/apis/upgrade.cattle.io/v1"
+	planClientset "github.com/rancher/system-upgrade-controller/pkg/generated/clientset/versioned/typed/upgrade.cattle.io/v1"
+	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// deployPlans creates a master and worker plan in the downstream cluster to instrument
+// the system-upgrade-controller in the downstream cluster
+func (h *handler) deployPlans(cluster *v3.Cluster) error {
+
+	// access downstream cluster
+	clusterCtx, err := h.manager.UserContext(cluster.Name)
+	if err != nil {
+		return err
+
+	}
+
+	// create a client for GETing Plans in the downstream cluster
+	planConfig, err := planClientset.NewForConfig(&clusterCtx.RESTConfig)
+	if err != nil {
+		return err
+	}
+	planClient := planConfig.Plans(metav1.NamespaceAll)
+
+	planList, err := planClient.List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	masterPlan := planv1.Plan{}
+	workerPlan := planv1.Plan{}
+	// deactivate all existing plans that are not managed by Rancher
+	for _, plan := range planList.Items {
+		if _, ok := plan.Labels[rancherManagedPlan]; !ok {
```

Comment:
Shouldn't the disablement of pre-existing plans, especially those not in the `cattle-system` namespace, be a one-time deal? Otherwise these are forever rendered unusable unless the user labels their nodes accordingly.

Reply:
As discussed, probably safe to leave this for now.

Reply:
captured in https://github.com/rancher/rancher/issues/25839